code/obj/machinery/alarm.dm

```diff
@@ -157,6 +158,17 @@
 	light_ov.plane = PLANE_SELFILLUM
 	src.UpdateOverlays(light_ov, "light")
 
+	var/list/cameras = list()
+	for_by_tcl(aiPlayer, /mob/living/silicon/ai)
+		if ((safe == ALARM_GOOD) && src.alertingAI)
+			aiPlayer.cancelAlarm("Atmosphere", get_area(src), src)
+			src.alertingAI = FALSE
+		else
+			for_by_tcl(C, /obj/machinery/camera)
+				if(get_area(C) == get_area(src))
+					cameras += C
+			aiPlayer.triggerAlarm("Atmosphere", get_area(src), cameras, src)
+			src.alertingAI = TRUE
```

Comment:
why are we running through every camera for every AI
should just be once for each